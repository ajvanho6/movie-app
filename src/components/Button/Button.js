import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './Button.scss';

const Button = ({className, children, ...rest}) => {
    const classes = classNames(
        'm-app-button',
        className,
    );

    return (
        <button type="button" className={classes} {...rest}>
            {children}
        </button>
    );
};

Button.propTypes = {
    className: PropTypes.string,
};

Button.defaultProps = {
    className: '',
};

export default Button;
