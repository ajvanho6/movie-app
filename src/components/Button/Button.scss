@import '../../styles/config';
@import '../../styles/mixins';

$component-class: m-app-button;

.#{$component-class} {
    position: relative;

    &--primary {
        border: 1px solid map-get($colors, background-6);
        color: map-get($colors, text-3);
        border-radius: 3px;
        padding: 8px;

        &:hover {
            cursor: pointer;
        }
    }

    &--fixed {
        position: fixed;
        top: 153px;
        right: 80px;
    }

    &--fixed-watch {
        top: 203px;
        min-width: 156px;
    }
}
