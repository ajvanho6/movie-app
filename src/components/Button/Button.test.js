import React from 'react';
import {shallow} from 'enzyme';
import Button from './Button';

describe('Button component', () => {
    it('Button should render correctly', () => {
        const component = shallow(<Button />);
        expect(component).toMatchSnapshot();
    });
    it('Test click event', () => {
        const mockCallBack = jest.fn();

        const button = shallow((<Button onClick={mockCallBack}>Ok!</Button>));
        button.find('button').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    });
});
