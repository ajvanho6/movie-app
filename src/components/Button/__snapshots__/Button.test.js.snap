// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Button should render correctly 1`] = `ShallowWrapper {}`;
